core/src/test/java/com/radixdlt/rev2/protocol/BottlenoseProtocolUpdateTest.java

```diff
@@ -0,0 +1,227 @@
+import com.radixdlt.statecomputer.RustStateComputer;
+import com.radixdlt.sync.TransactionsAndProofReader;
+import java.util.Arrays;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public final class BottlenoseProtocolUpdateTest {
+
+  private static final long BOTTLENOSE_EPOCH = 8;
+
+  private static final ProtocolConfig PROTOCOL_CONFIG =
+      new ProtocolConfig(
+          ImmutableList.of(
+              // Update to Anemone at some arbitrary earlier moment (in case of dependencies):
+              new ProtocolUpdateTrigger(
+                  ProtocolUpdateTrigger.ANEMONE, unconditionallyAtEpoch(BOTTLENOSE_EPOCH - 3)),
+              new ProtocolUpdateTrigger(
+                  ProtocolUpdateTrigger.BOTTLENOSE, unconditionallyAtEpoch(BOTTLENOSE_EPOCH))));
+
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+
+  private DeterministicTest createTest(Module... extraModules) {
+    return DeterministicTest.builder()
+        .addPhysicalNodes(PhysicalNodeConfig.createBatch(1, true))
+        .messageSelector(firstSelector())
+        .messageMutator(MessageMutator.dropTimeouts())
+        .addModules(extraModules)
+        .functionalNodeModule(
+            new FunctionalRadixNodeModule(
+                FunctionalRadixNodeModule.NodeStorageConfig.tempFolder(folder),
+                true,
+                FunctionalRadixNodeModule.SafetyRecoveryConfig.BERKELEY_DB,
+                FunctionalRadixNodeModule.ConsensusConfig.of(1000),
+                FunctionalRadixNodeModule.LedgerConfig.stateComputerNoSync(
+                    StateComputerConfig.rev2(
+                        Network.INTEGRATIONTESTNET.getId(),
+                        GenesisBuilder.createTestGenesisWithNumValidators(
+                            1,
+                            Decimal.ONE,
+                            GenesisConsensusManagerConfig.Builder.testWithRoundsPerEpoch(5)),
+                        StateComputerConfig.REV2ProposerConfig.Mempool.singleTransaction(),
+                        PROTOCOL_CONFIG))));
+  }
+
+  @Test
+  public void example_bottlenose_feature_is_available_only_after_update() throws ApiException {
+    // The easiest "new feature" to assert on is the AccountLocker package being published:
+    final var addressing = Addressing.ofNetwork(Network.INTEGRATIONTESTNET);
+    final var accountLockerCall =
+        new TransactionCallPreviewRequest()
+            .network(Network.INTEGRATIONTESTNET.getLogicalName())
+            .target(
+                new BlueprintFunctionTargetIdentifier()
+                    .packageAddress(
+                        addressing.encode(ScryptoConstants.ACCOUNT_LOCKER_PACKAGE_ADDRESS))
+                    .blueprintName("AccountLocker")
+                    .functionName("instantiate_simple")
+                    .type(TargetIdentifierType.FUNCTION))
+            .addArgumentsItem("4d0101"); // hex-encoded SBOR `true` (for `allow_recover` parameter)
```

Comment:
:D nice SBOR skillz! Side note, the TransactionCallPreviewRequest API could really do with an overhaul... I should raise a ticket for it at some point.

Reply:
> Is there an easy java-side Sbor.encode(true) that I could use here?
I think you be wanting `new ScryptoSbor(new CodecMap()).encode_payload(true)` - but I'm happy to leave it as `"4d0101"`.